### DISCUSSION TITLE
Forge Extension List and Extension Replacement List (Temporary)

### DISCUSSION THREAD

**Discussion by lllyasviel** [lllyasviel/stable-diffusion-webui-forge#1754] [Announcements]
Many important Automatic1111 extensions are not updated for months. 

Fortunately, some extensions are forked by new people.

However, it is difficult to find out which repo replaces which repo.

As a result, we recommend **all Forge extension makers to post a link to your extension here, if you replace some old extension, or build some new extension that only works in the latest Forge**.

People can search this page with “Ctrl+F”

Probably there will be a better solution for this indexing later.

---

Note:

1. Unrelated comments in this post will be removed.
2. Wrong info will be removed.
3. You do not need to be the author of the extension you report, but wrong info will be removed.

**Comment by lllyasviel** [comment]
Please copy this template when posting

---

Extension: sd-forge-xxxx
URL: https://github.com/xxx/xxx
Desc: xxxx

**Reply by lllyasviel** [reply]
hi people please use the global comment - do not reply my message

**Reply by derfasthirnlosenick** [reply]
Not the author but.

Extension: sd-forge-regional-promter
URL: https://github.com/jessearodriguez/sd-forge-regional-prompter
Desc: Allows you to set different prompts for different regions of your canvas. Original extension wasn too polished and a few things a re still broken in this fork, but key functionality seems to be there.

**Comment by lllyasviel** [comment]
Not sure but this seems the latest fork of deforum

https://github.com/Tok/sd-forge-deforum

**Comment by lllyasviel** [comment]
Not sure but this seems the latest fork of iclight
https://github.com/Haoming02/sd-forge-ic-light

**Comment by lllyasviel** [comment]
Not sure but this seems lastest ar-plus

https://github.com/altoiddealer/--sd-webui-ar-plusplus

**Reply by altoiddealer** [reply]
Name: `Aspect Ratio and Resolution Buttons`
Extension: `sd-webui-ar-plusplus`
URL: https://github.com/altoiddealer/--sd-webui-ar-plusplus
Desc: `Fork of Aspect Ratio selector plus with optimal resolution calculation and rounding`

A1111 screenshot
<img width="664" alt="Screenshot 2024-09-09 201752" src="https://github.com/user-attachments/assets/f1efa75a-3ba3-49b9-871f-c80626854bb2">

**Reply by altoiddealer** [reply]
The parent was also updated for Forge.

**Not obsolete:**

Extension: sd-webui-ar-plus
URL: https://github.com/LEv145/--sd-webui-ar-plus
Desc: `Fork of Aspect Ratio selector with new features and logic for aspect ratio`

<img width="645" alt="Screenshot 2024-09-09 202640" src="https://github.com/user-attachments/assets/86da4ea6-74fa-42ce-9d8a-98adf0277217">

**Comment by lllyasviel** [comment]
Also this is another list: https://github.com/lllyasviel/stable-diffusion-webui-forge/discussions/888
Help wanted if someone can help compile some items into this page so that people can `Ctrl+F`

**Comment by jordanjay29** [comment]
FaceswapLab: https://github.com/Ethereum-John/sd-webui-forge-faceswaplab

Fork mentioned on reddit.

**Comment by lvyonghuan** [comment]
sd-webui-segment-anything:https://github.com/altoiddealer/sd-webui-segment-anything-altoids/tree/work_in_forge
In `work_in_forge` branch.

**Reply by tedtninety** [reply]
Has anyone successfully used this in Forge UI?

**Reply by Q-King-8** [reply]
so do i，can't through controlnet inpaint

**Comment by Marc0ai** [comment]
Photo Refiner
https://github.com/Marc0ai/sd-webui-photo-refiner
Just a little extension I created to apply photo filters to generations.

**Comment by DeeepUx** [comment]
Supermerger
https://github.com/hako-mikan/sd-webui-supermerger.git

**Comment by AdrianoAz** [comment]
Extension: stable-diffusion-webui-GPU-temperature-protection
URL: https://github.com/w-e-w/stable-diffusion-webui-GPU-temperature-protection
Desc: Working fine.

I Run too much images for testing models in "Prompts from file or textbox" script, it's so fast im my RTX 3090, but my old cpu (an AMD FX(tm)-8120 Eight-Core Processor 3.10 GHz with 32Gb Ram) can't hold longer so much processing power and when the temperature reaches  about 70 celsius the pc turn off. in automatic 1111 the problem don't exist actually because it is more slow, giving time to processor don't overheat.
So I let the GPU temperature go till 69 celsius and let ir rest till 43, giving time to CPU be cool!

thanks, bye.

**Reply by baoanhng** [reply]
This is a cool thing

**Comment by DenOfEquity** [comment]
Extension: Cleaner
URL: https://github.com/DenOfEquity/forge2_cleaner
Description: a modified version of [novitalabs cleaner](https://github.com/novitalabs/sd-webui-cleaner), now uses Gradio4 image components

Extension: HyperTile
URL: https://github.com/DenOfEquity/HyperTile
Description: the old **HyperTile Integrated** extension, updated to work with Forge2. Faster SD1 and XL generations at the cost of some quality.

Extension: Multi-prompt (and Shift)
URL: https://github.com/DenOfEquity/forgeFlux_dualPrompt
Description: dual prompting for Flux and sdXL; and Shift for Flux (Simple scheduler only).

**Reply by maraan666** [reply]
Cleaner is fab! Thank you!

**Comment by Juqowel** [comment]
Extension: T5 on Other Device
URL: [https://github.com/Juqowel/GPU_For_T5](https://github.com/Juqowel/GPU_For_T5)
Desc: Moves the text encoder(CLIP/T5) to the selected device(CPU/CUDA) instead of the main GPU.

**Comment by zeittresor** [comment]
Extension: sd-test-prompts
URL: https://github.com/zeittresor/sd-test-prompts
Desc: Script Extension with some predefined prompts for A1111/Forge Stable Diffusion WebUI to test options

**Comment by Filexor** [comment]
Extension: sd-webui-top-k-emphasis
URL: https://github.com/Filexor/sd-webui-top-k-emphasis
Desc: Advanced method of emphasises which allows more strong conditioning or negative conditioning.

**Comment by CCpt5** [comment]
Extension: μ DDetailer
URL: https://github.com/wkpark/uddetailer
Desc: Fork of the same repo that Adetailer is based on (ddetailer). Uddetailer has similar functionality and an active developer (wpkpark) who will address compatibility issues w/ Forge if/when they are reported. (Mentioned in a thread re: feature not working in Forge: https://github.com/wkpark/uddetailer/issues/99)

**Comment by yeyugong** [comment]
[sd-webui-inpaint-anything](https://github.com/Uminosachi/sd-webui-inpaint-anything)
can not chose

**Reply by tedtninety** [reply]
I saw the same problem. It loads the colored map, but the mouse doe snot turn into a pencil to select any areas. :(

**Comment by supersonic13** [comment]
Extension: sd-webui-agent-scheduler
URL: https://github.com/supersonic13/sd-webui-agent-scheduler
Desc: Fork of the original agent scheduler (https://github.com/ArtVentureX/sd-webui-agent-scheduler) - updated to work with Gradio 4 and Forge due to the issues raised in the original repo (https://github.com/ArtVentureX/sd-webui-agent-scheduler/issues/256)

**Reply by van1027** [reply]
Doesn't seem to work... Clicking the Enqueue button and nothing happens. Is it platform dependent? I am on Ubuntu.

**Comment by evelryu** [comment]
Does anyone heave a working version of One Button Prompt for Forge?

**Comment by zeittresor** [comment]
Extension: sd-webui-mosaic-outpaint
URL: https://github.com/Haoming02/sd-webui-mosaic-outpaint
Desc:  Extension for the A1111 Webui, which trivializes outpainting via a mosaic pattern, without the need of ControlNet.
Info: (Not my extension). Extension works fine, Menu is in the top row, using it the mosaic / mask can be send to I2I for example.

**Comment by zeittresor** [comment]
Extension: Config-Presets
URL: https://github.com/Zyin055/Config-Presets
Desc: Extension adds a configurable dropdown to allow you to change settings in the txt2img and img2img tabs of the Web UI.
Info: (Not my extension). Extension works fine, the config options are below the preview window on the right side of the webui.

**Comment by zeittresor** [comment]
Extension: seamless-tile-inpainting
URL: https://github.com/brick2face/seamless-tile-inpainting
Desc: Script extension for making tiles seamless using inpainting.
Info: (Not my extension). Extension works fine, just go to img2img, drop a picture, select it under "scripts" to make a seamless tile.

**Comment by zeittresor** [comment]
Extension: sd_web_ui_scripts
URL: https://github.com/FartyPants/sd_web_ui_scripts
Desc: Some Scripts to simple swaping negative and positive prompts for giggles. Best with complicated prompts for more crazy results.
Info: (Not my extension). Extension works fine, select one of the scripts under Script Section of txt2img (i have only tested "Hallucinate").

**Comment by zeittresor** [comment]
Extension: advanced_euler_sampler_extension
URL: https://github.com/licyk/advanced_euler_sampler_extension
Desc: Additional Euler Samplers (pyramid noise) and more.
Info: (Not my extension). Extension works fine, Adds some more (different) Euler samplers in this extended fork.

**Comment by zeittresor** [comment]
Extension: sd-webui-tcd-sampler
URL: https://github.com/licyk/sd-webui-tcd-sampler
Desc: Adds two "TCD" Samplers
Info: (Not my extension). Extension works fine, simple select it from the samplers section.

**Comment by anapnoe** [comment]
Extension: sd-webui-ux
URL: https://github.com/anapnoe/sd-webui-ux
Desc: A bespoke, highly adaptable, blazing fast user interface that offers unmatched optimization and flexibility.
Info: This extension focuses solely on frontend optimizations.

**Reply by AnabelleKM** [reply]
Hi anapnoe,

I'm really glad to see something like this is possible. I was hoping for a different looking UI in the future. Mostly for the settings tab.

Now I've tried this extension, but at the moment I must admit it it's not too bad but it hurts my eyes. I have difficulties finding what I want. The text replacements of tabs to make them just icons isn't helping in my opinion. The extension tab is overwhelming to me, I need all extensions listed, not in boxes. In summary I kind of like it, but I guess not in it's current state. Still good work!

**Reply by anapnoe** [reply]
Hi @AnabelleKM, you can enable labels for the main , aside, and section tabs by adjusting the `Settings` > `Anapnoe UI-UX > User Interface`. Additionally, the theme configurator allows you to change the size, padding, color, borders etc and customize the style to your preference feel free to explore load diff styles for some ideas. Once you get the hang of it, you'll never want to go back!

**Reply by AnabelleKM** [reply]
Ah ok, I'll have another look then.

**Comment by AcademiaSD** [comment]
Extension: sd-webui-live-portrait
URL: https://github.com/dimitribarbot/sd-webui-live-portrait
Desc: allows to add a [LivePortrait](https://github.com/KwaiVGI/LivePortrait) tab to the original Stable Diffusion WebUI to benefit from LivePortrait features.
Info: Works perfectly in Forge, FFMPEG is required in the system

**Comment by Avaray** [comment]
Extension: **lora-keywords-finder**  
URL: https://github.com/Avaray/lora-keywords-finder
Description: Extension lets you find keywords for your [LoRA](https://wiki.civitai.com/wiki/Low-Rank_Adaptation) models using [CivitAI API](https://developer.civitai.com/docs/api/public-rest).

![Clipboard_10-26-2024_01](https://github.com/user-attachments/assets/6df1cc0f-baef-4a85-8ea4-359efbcc8344)

**Reply by bsdesigner** [reply]
Do you know where to rewrite the path to Lora? Because the extension sees only the path to the main folder with Lora, but does not read subfolders....
![Снимок экрана 2024-10-26 155656](https://github.com/user-attachments/assets/29402073-56f8-4dcc-8a13-fb08c34f30f7)

**Reply by Avaray** [reply]
Oh, I didn't think about it. I will make directory readings recursive in next update. Probably today or tomorrow.

**Reply by bsdesigner** [reply]
> Oh, I didn't think about it. I will make directory readings recursive in next update. Probably today or tomorrow.

Thank You!

**Reply by Avaray** [reply]
It's already done 👍 

If you encounter any issues, please open an issue [here](https://github.com/Avaray/lora-keywords-finder/issues)

**Reply by bsdesigner** [reply]
Super! Thanks for your work!

Upd: Unfortunately, as before, the extension only sees Lora files in the main folder. It doesn't see them in subfolders....

**Comment by Digiearts** [comment]
**Extension: sd-forge-surface-pen-fix**
URL: [https://github.com/Digiearts/sd-forge-surface-pen-fix](https://github.com/Digiearts/sd-forge-surface-pen-fix)

Description:
This extension is a fix for this issue: https://github.com/lllyasviel/stable-diffusion-webui-forge/issues/881 that I opened around 4 months ago.

I have developed an extension myself to fix this issue. 
With this extension the Microsoft Surface Pen will work smoothly again in latest SD Forge canvas with Gradio 4.
If you have similar issue with Microsoft Surface Pen you can install this extension to fix it.
Tested with Microsoft Surface Laptop Studio.

**Reply by Hujikuio** [reply]
Fixes canvas for Galaxy S-Pen as well.

**Reply by Digiearts** [reply]
Good to know. Hope this can be used as general fix for any devices using pen/ windows ink as well

**Reply by craggydaddy** [reply]
@Digiearts Is it possible for you to add a toggle for pressure sensitivity? I've noticed that when I do soft inpainting, I have to press really hard to completely inpaint, and there's no visual indicator as to what opacity the current mask is at.

**Reply by Digiearts** [reply]
> @Digiearts Is it possible for you to add a toggle for pressure sensitivity? I've noticed that when I do soft inpainting, I have to press really hard to completely inpaint, and there's no visual indicator as to what opacity the current mask is at.

Hi thanks for the comment. I don't see issue with pressure on my Surface laptop studio. What tablet do you use? I will see the possibility to add pressure sensitivity toggle.

**Reply by craggydaddy** [reply]
> > @Digiearts Is it possible for you to add a toggle for pressure sensitivity? I've noticed that when I do soft inpainting, I have to press really hard to completely inpaint, and there's no visual indicator as to what opacity the current mask is at.
> 
> Hi thanks for the comment. I don't see issue with pressure on my Surface laptop studio. What tablet do you use? I will see the possibility to add pressure sensitivity toggle.

I think it was either a conflicting extension with gradio 4 canvas, or something else as it seems to have resolved itself. I'm using a S24 Ultra with a S-Pen, but I think the Galaxy Tabs are all the same as well. Regardless, since it doesn't seem to be the issue I wouldn't want you to waste your time on that feature, especially since photopea exists which has pressure toggle built in already. Thank you for figure out how to make these pens work on Gradio 4/Forge though, it's the only reason I'm not using reforge (which of course has a whole host of its own canvas problems, but at least it was somewhat usable VS stock forge canvas)

**Comment by Digiearts** [comment]
**Extension: sd-forge-cleaner**
URL: [https://github.com/Digiearts/sd-forge-cleaner](https://github.com/Digiearts/sd-forge-cleaner)

Description:
This extension is a modification of the extension [sd-webui-cleaner](https://github.com/novitalabs/sd-webui-cleaner) originally developed by **novitalabs**. This old extension does not work anymore in latest SD Forge with Gradio 4 environment.

With this renewed / modified extension, all the cleaner functionality and canvas element work again with latest SD Forge environment.

**Reply by UnconnectedBedna** [reply]
Please remove the dot (.) in the link to original project.
Also, why is it not a fork?

**Reply by Digiearts** [reply]
I just removed the dot. Thanks for the notice. I did initially the fixes directly from the plugin I installed in my SD forge for my personal use, then decided to share it if someone else may need it.

**Reply by UnconnectedBedna** [reply]
I'm not sure that is 100% ok, but I am no license expert...

https://github.com/novitalabs/sd-webui-cleaner?tab=MIT-1-ov-file
> Copyright (c) 2023 novitalabs
>
> The above copyright notice and this permission notice shall be included in all
copies or substantial portions of the Software.

Just make sure you are not stepping on anybody's toes here... :)

If you want to be 100% sure, just fork the project and make the changes to that repo.

**Reply by Digiearts** [reply]
Thanks for your notes! I am also not a license expert. I just throw the question to a GPT bot about the essence of an MIT license, and got this answer. But really thanks anyway for your concern and notification.
 
Yes, you can copy and modify the code without forking the original GitHub repository. This is the MIT License, one of the most permissive open source licenses. Here's what it allows you to do:

Copy the code directly
Modify it as you wish
Use it commercially
Distribute it
Sublicense it
Sell it
The only requirements are:

You must include the original copyright notice
You must include the MIT license text in your copy/modified version

You don't need to:
Fork the repository
Ask for permission
Make your modifications public
Pay any fees
Notify the original authors
The license essentially says "do whatever you want with this code, just keep the license and copyright notice, and we're not responsible for any problems that might occur."

This is why the MIT License is very popular in open source - it provides maximum freedom to users while protecting the original authors from liability.

**Reply by UnconnectedBedna** [reply]
> You must include the original copyright notice
> You must include the MIT license text in your copy/modified version

And you did not do that now did you?
To me it looks like you created your own MIT license.

You do you, I just feel it is a bit disrespectful not forking to show origin.

**Edit**
https://github.com/Digiearts/sd-forge-cleaner/compare/174b1c1f4e56e0ac2e8d40fdbdf3607944af7361...79bcce8debcd867ea8b73bf51c0cecc3a0fc5338

I think you are good now. :)

**Reply by Digiearts** [reply]
Yes did not really notice that. the file was automatically created by GitHub when uplodaing the repo for the first time. Just updated. Thanks ;-)

**Reply by UnconnectedBedna** [reply]
That's what I was suspecting happened, and also the reason I recommended you fork the project and then make changes, but this works too.

NOW I feel ok with testing the plugin. :)
Thanks for your FOSS contribution! :heart:

**Comment by altoiddealer** [comment]
Please replace the Segment Anything extension in the Extension list with my fork.

I had submitted the updates to the main Segment Anything project, but the dev has been on hiatus for months.

Name: Segment Anything
Extension: `sd-webui-ar-segment-anything-altoids`
URL: https://github.com/altoiddealer/sd-webui-segment-anything-altoids
Desc: `Fork of Segment Anything updated to work in both Gradio 3 and 4. Automatically generate high-quality segmentations/masks for images by clicking or text prompting. Aim for connecting WebUI and ControlNet with Segment Anything and GroundingDINO to enhance Stable Diffusion/ControlNet inpainting (both single image and batch process), enhance ControlNet semantic segmentation, automate image matting and create LoRA/LyCORIS training set.`

Thanks!

**Reply by aboghalon** [reply]
Thanks for this. It works except for the image and mask not copying over to inpaint upload when the box is checked.

**Reply by overallbit** [reply]
Please @altoiddealer could you fix it one more?

**Reply by altoiddealer** [reply]
I’ll take a look and see if I can figure it out!  Will reply here with an update soon

**Reply by altoiddealer** [reply]
Here's my reply - I'm pretty OK with python, but struggle a lot with gradio code (which is in python), but have almost no knowledge of javascript.  And the code that is supposed to execute here is a combination of javascript and gradio.

[Here's the code for the button click](https://github.com/altoiddealer/sd-webui-segment-anything-altoids/blob/6cb6c64ba7f6eac5bd42e8fb87e035886af537d9/scripts/sam.py#L788-L792)
[Here's the javascript that is executed](https://github.com/altoiddealer/sd-webui-segment-anything-altoids/blob/6cb6c64ba7f6eac5bd42e8fb87e035886af537d9/javascript/sam.js#L19-L21)

I'm not particularly interested in trying to debug this, it looks like I'd have to research quite a bit and pull out some hair to add the mild convenience of sending the images instead of saving them then drag and dropping them.  Sorry :(

**Comment by zeittresor** [comment]
Extension: sd-forge-fum
URL: https://github.com/zeittresor/sd-forge-fum
Desc: Replacement for the BuildIn FreeU Implementation in Forge with additional options for animations

**Comment by licyk** [comment]
Extension: sd_forge_hypertile_svd_z123
URL: https://github.com/licyk/sd_forge_hypertile_svd_z123
Desc: add hypertile, svd and z123 in webui / forge

**Comment by tocantrell** [comment]
Extension: sd-refdrop-forge
URL: [https://github.com/tocantrell/sd-refdrop-forge](https://github.com/tocantrell/sd-refdrop-forge)
Desc: RefDrop image consistency based on [RefDrop: Controllable Consistency in Image or Video Generation via Reference Feature Guidance](https://arxiv.org/abs/2405.17661) (Fan et al. 2024). It allows for either consistency or diversification of diffusion model outputs based on an initially recorded output. The level of consistency or diversification is controlled by the RFG Coefficent, which is a parameter that ranges from -1 to 1. Positive values force the outputs to be similar to the initial image while negative values ensure differences.

**Reply by hongkong7530** [reply]
**That's awesome! Thank you!**

**Comment by zhaol4v** [comment]
Extension: auto-noise-schedule
URL: https://github.com/michP247/auto-noise-schedule
Desc: Better method of ensuring that you are always using the recommended noise schedule for the loaded Epsilon-pred or V-Pred model without manual intervention.

**Comment by RegulusAlpha** [comment]
Extension: sd-webui-watermark
URL: https://github.com/RegulusAlpha/sd-webui-watermark
This extension automatically adds a customizable text or image watermark to images saved in Stable Diffusion WebUI Forge, and provides a UI tab for batch watermarking existing images.

**Comment by ATata-name** [comment]
Extension: sd-webui-inpaint-noobaixl-preprocessor
URL: https://github.com/ATata-name/sd-webui-inpaint-noobaixl-preprocessor.git
Desc: Preprocessor for WebUI Forge to work with inpainting ControlNet for Noobai XL.

**Comment by LeDXIII** [comment]
Extension: SADA-Extension-for-Forge-WebUI
URL: https://github.com/LeDXIII/SADA-Extension-for-Forge-WebUI
Desc: This repository implements SADA: Stability-guided Adaptive Diffusion Acceleration for ForgeUI. Speeds up SD XL, Flux by 1.3-2.0 times with almost no quality loss.

**Reply by AnabelleKM** [reply]
Very nice! Thank you, it works.

**Comment by RegulusAlpha** [comment]
Extension: Dynamic Prompting Simplified
URL: https://github.com/RegulusAlpha/dynamic-prompting-simplified

Desc: I wanted a simpler dynamic prompting extension that supported mirrored prompts in the positive and negative prompt. wildcard files with -mir tag will mirror in the positive and negative prompt

example:
{red hat|blue hat|green hat} in hats-mir.txt
positive picks 'red hat'
negative will mirror it with blue hat, green hat

otherwise dynamic prompting will work as expected with nesting support {red hat|blue hat|{green hat|{yellow hat|orange hat}}}
 
this extension does not have all the capabilities as sd-dynamic-prompts and is a stripped down simplified version.

**Comment by haasinc42** [comment]
Extension: sd-webui-inpaint-anything-forge
URL: https://github.com/thadius83/sd-webui-inpaint-anything-forge

Desc: This is a specialized fork of the Inpaint Anything extension specifically optimized for [SD WebUI Forge](https://github.com/lllyasviel/stable-diffusion-webui-forge) with Gradio ≥ 4.40.
---------------
 not the author, just mention it here in case anyone wonder the compatibility of inpaint anything

**Comment by baoanhng** [comment]
Anything to save last state?

**Reply by zeittresor** [reply]
hi there.. IMHO not "directly" but you can save the "last state" using the "Config-Presets" extension..

**Reply by MrKuenning** [reply]
https://github.com/ilian6806/stable-diffusion-webui-state.git

**Comment by zeittresor** [comment]
Extension: sd-webui-tcd-sampler
URL: https://github.com/zeittresor/sd-webui-tcd-sampler
Desc: Fork of https://github.com/licyk/sd-webui-tcd-sampler with some additional TCD Samplers

**Comment by megvadulthangya** [comment]
Extension: Stylez
URL: https://github.com/megvadulthangya/Stylez
Desc: (This is fork of Stylez)An improved styles Library with 600+ built in stlyes. Prompt generation is included. Style creating/Editing. Auto CSV conversion to work with Stylez. CivitAI image browser.

FIxed works in gradio 4.40 (plus removed unnessary tab)

**Comment by megvadulthangya** [comment]
*update: merged into the original project, so you can use the original project:

https://github.com/navimixu/wildcard-gallery

Extension: wildcard-gallery

_URL: https://github.com/megvadulthangya/wildcard-gallery.git_

Desc: A modern fork of Wildcard Gallery V2, fully updated for Gradio 4.40+ and SD WebUI Forge compatibility. It fixes the broken UI tabs, removes the restrictive whitelist logic, and implements recursive directory scanning for large wildcard collections.

___

I need feedback on my repo pls!

_Im not sure about this, becouse i never see the working 2.0 verzion, i only used v1 wildcard gallery which was simple, So i guess i need some feedback about this, i may remove functions what i dont understand, or I didn't have the nerve to understand the original dev's intentions._

Or OG dev ( @navimixu ) feedback would be the best, if its have time to check it, what mess i do, and what needs to restore, couse i want a fast fix, and removed whitelist, which was annoying, at that time i have 3000+ wildcards and dont want to mess with whitelist...

**Reply by navimixu** [reply]
will do 👍, thanks for your time .

**Comment by Dragynrain** [comment]
**Asymmetric Tiling - Forge-compatible fork**

  The popular https://github.com/tjm35/asymmetric-tiling-sd-webui (214 stars) doesn't work with Forge (https://github.com/tjm35/asymmetric-tiling-sd-webui/issues/16, closed as won't-fix) - it errors out because Forge removed the model_hijack.layers API it depends on. I've published a rewrite that works with Forge while keeping A1111 compatibility.

  Install URL: https://github.com/Dragynrain/asymmetric-tiling-sd-webui.git

  (Extensions tab -> Install from URL -> paste -> Install -> restart)

  What changed:

  1. Uses Forge's forge_objects.unet / forge_objects.vae to access model layers instead of the removed model_hijack.layers
  2. Uses isinstance() instead of type() == Conv2d, since Forge uses a custom Conv2d subclass that exact matching silently skips
  3. Wraps forward() instead of _conv_forward(), since Forge's Conv2d has a code path that bypasses instance-level _conv_forward patches
  4. Patches the VAE separately - without this you get faint seam lines in the decoded image

  Tested with SDXL on Forge f2.0.1v1.10.1. CC0 licensed, same as the original.

**Comment by Gunther-Schulz** [comment]
### sd-webui-prompt-enhancer

**Repo:** https://github.com/Gunther-Schulz/sd-webui-prompt-enhancer

Local LLM-powered prompt builder for Forge/Forge Classic/A1111. Uses Ollama to generate detailed prompts or booru-style tags from short descriptions.

**Key features:**
- Three modes: Prose (flowing paragraph), Tags (Illustrious/NoobAI/Pony booru tags), Remix (modify existing)
- 130+ categorized modifiers across 6+ dropdowns (Subject, Setting, Lighting & Mood, Visual Style, Camera, Audio)
- Tag validation with auto-downloaded danbooru tag databases, alias correction, fuzzy matching
- Wildcards for creative LLM choices (surprise location, random artist, anime era, etc.)
- Detail slider that scales to the active model's token limit (SD/SDXL/Flux/Z-Image)
- Local YAML overrides — each file becomes a dropdown, extend without touching published code
- Extensible tag formats — add new model support by dropping a YAML file
- Streaming with stall/thinking detection, cancel button
- Zero VRAM impact (Ollama on CPU by default)

Recommended model: `huihui_ai/qwen3.5-abliterated:9b`

Built with [Claude Code](https://claude.ai/claude-code) (Anthropic).

**Comment by therestlesscat** [comment]
**sd-webui-forge-model-manager**

Repository: https://github.com/therestlesscat/sd-webui-forge-model-manager.git

I've been working on a **Model Manager extension for SD WebUI Forge Neo** that provides Civitai integration and tools for working with your local model library directly from the WebUI.

The extension was originally built for [SD WebUI Forge](https://github.com/lllyasviel/stable-diffusion-webui-forge) for my own private use. I recently updated it for **SD WebUI Forge Neo** and decided to make it public, while keeping backward compatibility with the original SD WebUI Forge.

### Quick Showcase

https://github.com/user-attachments/assets/185cead5-0e5a-4d61-9561-e5420c84e31b

[Watch it in better quality on YouTube](https://youtu.be/HCHe44Yp-QU)

The project is still actively being worked on, so feedback, bug reports, and feature suggestions are very welcome.
